src/protocol/frame/mod.rs

```diff
@@ -233,8 +234,9 @@ impl FrameCodec {
     fn read_in(&mut self, stream: &mut impl Read) -> io::Result<usize> {
         let len = self.in_buffer.len();
         debug_assert!(self.in_buffer.capacity() > len);
-        self.in_buffer.resize(self.in_buffer.capacity().min(len + self.in_buf_max_read), 0);
-        let size = stream.read(&mut self.in_buffer[len..]);
+        let longer_len = (len + self.in_buf_max_read).min(self.in_buffer.capacity());
+        self.in_buffer.resize(longer_len);
+        let size = stream.read(&mut self.in_buffer[len..longer_len]);
```

Comment:
We could switch back to providing the entire capacity to `Read` now we can init that efficiently. However, it isn't clear what advantage that would provide exactly and I think continuing to use the `read_buffer_size` as the max give users more control over the behaviour. It's possible some `Read` impls might not handle larger buffers well too :shrug:.